ffmpeg/lpms_ffmpeg.c

```diff
@@ -753,6 +797,37 @@ static int open_video_decoder(input_params *params, struct input_ctx *ctx)
       AVCodec *c = avcodec_find_decoder_by_name("h264_cuvid");
       if (c) codec = c;
       else fprintf(stderr, "Cuvid decoder not found; defaulting to software\n");
+    } else if (AV_CODEC_ID_H264 == codec->id &&
+                       AV_HWDEVICE_TYPE_VAAPI == params->hw_type) {
+      enum AVHWDeviceType type;
+      type = av_hwdevice_find_type_by_name("vaapi");
+      if (type == AV_HWDEVICE_TYPE_NONE) {
+        fprintf(stderr, "Device type vaapi is not supported.\n");
+        fprintf(stderr, "Available device types:");
+        while((type = av_hwdevice_iterate_types(type)) != AV_HWDEVICE_TYPE_NONE)
```

Comment:
Logging this bit of information may be helpful somewhere, but not sure if this here is the best place to do it.

Reply:
No longer required, `av_hwdevice_find_type_by_name("vaapi")` would always return `AV_HWDEVICE_TYPE_VAAPI` and subsequent check would always be false